packages/contracts/contracts/libs/v0.8.x/minter-libs/RAMLib.sol

```diff
@@ -0,0 +1,2926 @@
+     * @param minterRefundGasLimit Gas limit to use when refunding the bidder.
+     */
+    function directMintTokensToWinners(
+        uint256 projectId,
+        address coreContract,
+        uint32[] calldata bidIds,
+        bool requireSenderIsBidder,
+        IMinterFilterV1 minterFilter,
+        uint256 minterRefundGasLimit
+    ) internal {
+        // load project config
+        RAMProjectConfig storage RAMProjectConfig_ = getRAMProjectConfig({
+            projectId: projectId,
+            coreContract: coreContract
+        });
+
+        // get project price
+        uint256 projectPrice = _getProjectPrice({
+            RAMProjectConfig_: RAMProjectConfig_
+        });
+
+        // CHECKS
+        // @dev memoize length for gas efficiency
+        uint256 bidIdsLength = bidIds.length;
+        // @dev block scope to limit stack depth
+        {
+            // require project minter state D (Post-Auction,
+            // post-admin-artist-only, not all bids handled)
+            ProjectMinterStates projectMinterState = getProjectMinterState({
+                projectId: projectId,
+                coreContract: coreContract
+            });
+            require(
+                projectMinterState == ProjectMinterStates.PostAuctionOpenMint,
+                "Only post-auction open mint"
+            );
+            // require numTokensToMint does not exceed number of tokens
+            // owed.
+            // @dev must check this here to avoid minting more tokens than max
+            // invocations, which could potentially not revert if minter
+            // max invocations was limiting (+other unexpected conditions)
+            require(
+                bidIdsLength <=
+                    _getNumTokensOwed({RAMProjectConfig_: RAMProjectConfig_}),
+                "tokens to mint gt tokens owed"
+            );
+        }
+
+        // main loop to mint tokens
+        for (uint256 i; i < bidIdsLength; ++i) {
+            // @dev current slot index and bid index in slot not memoized due
+            // to stack depth limitations
+            // get bid
+            uint256 currentBidId = bidIds[i];
+            Bid storage bid = RAMProjectConfig_.bids[currentBidId];
+            // CHECKS
+            // if bid is already minted or refunded, skip to next bid
+            // @dev do not revert, since this could be due to front-running
+            if (
+                _getBidPackedBool(bid, INDEX_IS_MINTED) ||
+                _getBidPackedBool(bid, INDEX_IS_REFUNDED)
+            ) {
+                continue;
+            }
+            address bidderAddress = bid.bidder;
+            // require sender is bidder if requireSenderIsBidder is true
+            if (requireSenderIsBidder) {
+                require(msg.sender == bidderAddress, "Only sender is bidder");
+            } else {
+                // require bid exists
+                require(bidderAddress != address(0), "invalid Bid ID");
```

Comment:
I think this would be better to put before the `continue` operation, because I think reverting when an invalid Bid ID is supplied is a better UX than just skipping, and would help a mislead admin properly configure their tx. It comes at a slight gas increase, but probably worth it imo. thoughts?